### PR TITLE
User management refrac

### LINKED ISSUE #26: User Management
Make a command that has sub commands and can modify user data.

e.g
```
/usermanage view - shows data on user
/usermanage modify <credits> <permissionLevel> <blacklisted> - modifies specified user data
/usermanage purge - purges user orders
/usermanage resetdaily - reset daily timer
```

### PR DESCRIPTION
**Pull request by refractored**
resolves #26 
bandages #31

### CODE REVIEW

File: src/commands/blacklist.go [hunk 1]
```diff
@@ -37,11 +61,56 @@ func (c BlacklistCommand) permissionLevel() models.UserPermissionLevel {
...
+		})
+	}
+
+	options := event.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options[0].Options))
+	for _, opt := range options[0].Options {
+		optionMap[opt.Name] = opt
+	}
+	switch options[0].Name {
+
```
Comment (Baconing) [inline src/commands/blacklist.go]: remove for formatting

File: src/commands/command.go [hunk 2]
```diff
@@ -15,6 +15,9 @@ type Command interface {
 	// Returns the command data
 	getCommandData() *discordgo.ApplicationCommand
 
+	// Returns true if the command is allowed in DMs.
+	DMsAllowed() bool
```
Comment (Baconing) [inline src/commands/command.go]: ?

File: src/commands/bot.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+func (c BotCommand) registerGuild() string {
+	return config.GetConfig().GuildID
+}
+
+func (c BotCommand) permissionLevel() models.UserPermissionLevel {
+	return models.PermissionLevelAdmin
+}
+
+func (c BotCommand) execute(session *discordgo.Session, event *discordgo.InteractionCreate) {
+
```
Comment (Baconing) [inline src/commands/bot.go]: remove for formatting

File: src/commands/info.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+		},
+	}
+}
+
+func (c InfoCommand) DMsAllowed() bool {
+	return true
+}
+
+func (c InfoCommand) registerGuild() string {
+	return ""
```
Comment (Baconing) [inline src/commands/info.go:46]: admin commands should not be global

File: src/commands/status.go [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+func (c StatusCommand) registerGuild() string {
+	return ""
+}
+
+func (c StatusCommand) permissionLevel() models.UserPermissionLevel {
+	return models.PermissionLevelArtist
+}
+
+func (c StatusCommand) execute(session *discordgo.Session, event *discordgo.InteractionCreate) {
+
```
Comment (Baconing) [inline src/commands/status.go]: remove formatting

File: src/commands/usermanage.go [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+		},
+	}
+}
+
+func (c UserManageCommand) DMsAllowed() bool {
+	return true
+}
+
+func (c UserManageCommand) registerGuild() string {
+	return ""
```
Comment (Baconing) [inline src/commands/usermanage.go]: should not be global

File: src/commands/usermanage.go [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+			},
+		})
+		break
+	case "addcredits":
+		UserManageAddCredits(session, event)
+		break
+	case "takecredits":
+		UserManageTakeCredits(session, event)
+		break
+
```
Comment (Baconing) [inline src/commands/usermanage.go]: remove formatting

**Review by Baconing — CHANGES_REQUESTED** [review]
mostly small things

**Comment by Baconing** [comment]
wont resolve #31, I was thinking of a different solution for better readability and implementation, but this works as a bandaid solution

File: src/commands/info.go (hunk 4 above)
Comment (refractored) [reply src/commands/info.go:46]: This actually was not supposed to be an admin command, it is fixed now.

**Review by Baconing — APPROVED** [review]

**Pull request merged**